### PR TITLE
kubernetes plugin adds support for the new plugin system

### PR DESCRIPTION
**Pull request by RobotSail**
Signed-off-by: Oleg S <97077423+RobotSail@users.noreply.github.com>

## Hey, I just made a Pull Request!

This PR adds support for the new plugin system to the kubernetes-backend plugin.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by RobotSail** [comment]
This PR is part of #13084.

**Review by jamieklassen — APPROVED** [review]
LGTM, though at some point it might be nice to have a few extension points:
* [cluster supplier](https://github.com/backstage/backstage/blob/270529ff2f196daec626b04b3165d8cdc3a3b3cf/plugins/kubernetes-backend/src/service/KubernetesBuilder.ts#L142-L145)
* [service locator](https://github.com/backstage/backstage/blob/270529ff2f196daec626b04b3165d8cdc3a3b3cf/plugins/kubernetes-backend/src/service/KubernetesBuilder.ts#L162-L165)

File: .changeset/stale-cobras-beg.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-kubernetes-backend': minor
+'example-backend-next': minor
```
Comment (Rugvip) [inline .changeset/stale-cobras-beg.md:3]:
```suggestion
'@backstage/plugin-kubernetes-backend': minor
```

No need for changesets for internal packages

File: plugins/kubernetes-backend/api-report.md [hunk 2]
```diff
@@ -336,6 +337,9 @@ export type KubernetesObjectTypes =
   | 'statefulsets'
   | 'daemonsets';
 
+// @public
+export const kubernetesPlugin: () => BackendFeature;
```
Comment (Rugvip) [inline plugins/kubernetes-backend/api-report.md:341]: I think it's best to export this under `/alpha` like many other plugins have done tbh

File: plugins/kubernetes-backend/src/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        config: coreServices.config,
+        catalogApi: catalogServiceRef,
+      },
+      async init({ http, logger, config, catalogApi }) {
+        const winstonLogger = loggerToWinstonLogger(logger);
+        const { router } = await KubernetesBuilder.createBuilder({
+          logger: winstonLogger,
+          config,
+          catalogApi,
+        }).build();
```
Comment (Rugvip) [inline plugins/kubernetes-backend/src/plugin.ts:46]: As Jamie mentioned there's quite a bit more to getting this fully moved to the new system. Np to leave that for followup though, but perhaps we can add a `// TODO: expose all of the customization points of the builder` here

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Just a few nits

File: plugins/kubernetes-backend/api-report.md (hunk 2 above)
Comment (RobotSail) [reply plugins/kubernetes-backend/api-report.md:341]: Sounds like a plan

**Review by Rugvip — CHANGES_REQUESTED** [review]
👍 , we've ended up with some duplicate changesets though 😅

**Comment by RobotSail** [comment]
@Rugvip I've removed the duplicate changeset, was there anything else that you needed from this PR?

**Comment by RobotSail** [comment]
Not stale

**Review by Rugvip — APPROVED** [review]
Sorry, let's :shipit:!

**Comment by RobotSail** [comment]
Updated the PR, should be good to merge now

**Pull request merged**